File: 38_Count-and-Say.java
class Solution {
    public static String rle(String partial){
        StringBuilder result = new StringBuilder();
        int count = 1;
        for(int i = 1; i < partial.length();i++){
            if(partial.charAt(i-1)==partial.charAt(i)){
                    count++;
                }else{
                    result.append(count);
                    result.append(partial.charAt(i-1));
                    count = 1;
                }
            }
            result.append(count); 
            result.append(partial.charAt(partial.length()-1));
            return result.toString();
    }
    public static String recurse(int n){
        if(n == 1){
            return "1";
        }
        return rle(recurse(n-1));
    }
    public String countAndSay(int n) {
        return recurse(n);
    }
}

File: 38_Count-and-Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        def rle(partial):
            result = ""
            count = 0
            prev = False
            for i in partial:
                if prev:
                    if prev == i:
                        count += 1
                        prev = i
                    else:
                        result += str(count) + prev
                        prev = i
                        count = 1
                else:
                    prev = i
                    count += 1
            result += str(count) + prev 
            return result           

        def recurse(n):
            if n == 1:
                return "1"
            return rle(recurse(n-1))
        return recurse(n)
